src/components/views/elements/TagTile.js

```diff
@@ -22,17 +22,36 @@ import sdk from '../../../index';
 import dis from '../../../dispatcher';
 import { isOnlyCtrlOrCmdKeyEvent } from '../../../Keyboard';
 
+import FlairStore from '../../../stores/FlairStore';
+
 export default React.createClass({
     displayName: 'TagTile',
 
     propTypes: {
-        groupProfile: PropTypes.object,
+        tag: PropTypes.string,
     },
 
     contextTypes: {
         matrixClient: React.PropTypes.instanceOf(MatrixClient).isRequired,
     },
 
+    componentWillMount() {
+        this.unmounted = false;
+        if (this.props.tag[0] === '+') {
+            FlairStore.getGroupProfileCached(
```

Comment:
we're going to need some exception handling in here, otherwise an error will take out the whole of react

Reply:
Actually I think it'll just be an unhandled rejected promise, but we should still handle it.

Reply:
Although, there isn't really a graceful way to recover short of retrying, which seems OTT.